python/main.py

```diff
@@ -332,9 +335,361 @@ def recognized(evt):
         print(f"       {idx + 1:02d}:\t{word.word:16s}{word.accuracy_score:.1f}\t\t{phoneme_accuracy_score}")
 
 
+def read_wave_header(file_path):
+    with wave.open(file_path, 'rb') as audio_file:
+        framerate = audio_file.getframerate()
+        bits_per_sample = audio_file.getsampwidth() * 8
+        num_channels = audio_file.getnchannels()
+        return framerate, bits_per_sample, num_channels
+
+
+def push_stream_writer(stream, filenames):
+    # The number of bytes to push per buffer
+    n_bytes = 3200
+    try:
+        for filename in filenames:
+            wav_fh = wave.open(filename)
+            # Start pushing data until all data has been read from the file
+            try:
+                while True:
+                    frames = wav_fh.readframes(n_bytes // 2)
+                    # print('read {} bytes'.format(len(frames)))
+                    if not frames:
+                        break
+                    stream.write(frames)
+                    time.sleep(.1)
+            finally:
+                wav_fh.close()
+    finally:
+        stream.close()
+
+
+def chatting_from_file():
+    """Performs pronunciation assessment asynchronously from an audio file and enable prosody/content assessment
+        with continuous mode.
+        See more information at https://aka.ms/csspeech/pa"""
+
+    topic = "YourOwnTopic"
+    input_files = ["../resources/chat_input_1.wav", "../resources/chat_input_2.wav"]
+
+    def stt(filename):
+        result_text = []
+
+        speech_config = speechsdk.SpeechConfig(subscription=speech_key, region=service_region)
+        audio_config = speechsdk.audio.AudioConfig(filename=filename)
+
+        speech_recognizer = speechsdk.SpeechRecognizer(speech_config=speech_config, audio_config=audio_config)
+
+        done = False
+
+        def stop_cb(evt: speechsdk.SessionEventArgs):
+            """callback that signals to stop continuous recognition upon receiving an event `evt`"""
+            nonlocal done
+            done = True
+
+        # Connect callbacks to the events fired by the speech recognizer
+        speech_recognizer.recognized.connect(lambda evt: result_text.append(evt.result.text))
+        # Stop continuous recognition on either session stopped or canceled events
+        speech_recognizer.session_stopped.connect(stop_cb)
+        speech_recognizer.canceled.connect(stop_cb)
+
+        # Start continuous speech recognition
+        speech_recognizer.start_continuous_recognition()
+        while not done:
+            time.sleep(.5)
+
+        speech_recognizer.stop_continuous_recognition()
+        text = "".join(result_text)
+        print("YOU: ", text)
+        return text
+
+
+    def call_gpt(send_text):
+        your_resource_name = "YourResourceName"
```

Comment:
Please add the corresponding doc in the readme file.

Reply:
added